nomad/core_sched_test.go

```diff
@@ -1843,6 +1844,16 @@ func TestAllocation_GCEligible(t *testing.T) {
 			ThresholdIndex: 90,
 			ShouldGC:       false,
 		},
+		{
+			Desc:             "GC when terminal but not failed ",
```

Comment:
Should you add that it isn't being GC'd because the modify index is greater than the threshold? Reading the description confused me why `ShouldGC: false` on this one.

Reply:
copy paste bug in the test, fixed now, I meant to test that it should GC